File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/UrlCopyService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import com.windea.demo.cloudcollect.core.domain.response.*

interface UrlCopyService {
	/**复制为Html链接。*/
	fun copyAsHtmlUrl(collect: Collect): Url
	
	/**复制为Markdown链接。*/
	fun copyAsMdUrl(collect: Collect): Url
	
	/**复制为Markdown引用连接。*/
	fun copyAsMdRefUrl(collect: Collect): Url
	
	/**复制为Markdown图片链接。*/
	fun copyAsMdImgUrl(collect: Collect): Url
	
	/**复制为Markdown图片引用连接。*/
	fun copyAsMdImgRefUrl(collect: Collect): Url
}

File: cloud-collect-web/src/service/urlCopyService.ts
import http from "@/http"
import {Collect, Pair} from "@/types"

//NOTE 是否可考虑直接在前端实现这一功能？

const urlCopyUrl = "/urlCopy"

export async function copyAsHtmlUrl(collect: Collect) {
  return await http.post<string>(`${urlCopyUrl}/copyAsHtmlUrl`, collect)
}

export async function copyAsMdUrl(collect: Collect) {
  return await http.post<string>(`${urlCopyUrl}/copyAsMdUrl`, collect)
}

export async function copyAsMdRefUrl(collect: Collect) {
  return await http.post<Pair<string, string>>(`${urlCopyUrl}/copyAsMdRefUrl`, collect)
}

export async function copyAsMdImgUrl(collect: Collect) {
  return await http.post<string>(`${urlCopyUrl}/copyAsMdImgUrl`, collect)
}

export async function copyAsMdImgRefUrl(collect: Collect) {
  return await http.post<Pair<string, string>>(`${urlCopyUrl}/copyAsMdImgRefUrl`, collect)
}
